hibernate-core/src/test/java/org/hibernate/orm/test/mapping/hhh15842/EnumForeignKeyParameterMappingTest.java

```diff
@@ -0,0 +1,195 @@
+import jakarta.persistence.Table;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Jpa(annotatedClasses = {
+		EnumForeignKeyParameterMappingTest.EntityWithEnumPrimaryKey.class,
+		EnumForeignKeyParameterMappingTest.EntityWithEnumForeignKey.class
+})
+@TestForIssue(jiraKey = "HHH-15842")
+public class EnumForeignKeyParameterMappingTest {
+
+	List<EntityWithEnumPrimaryKey> entitiesWithEnumPk = new ArrayList<>();
+
+	@BeforeEach
+	public void setUp(EntityManagerFactoryScope scope) {
+		scope.inTransaction( em -> Arrays.stream( EnumForeignKeyParameterMappingTest.EnumPK.values() )
+				.forEach( enumVal -> {
+					EntityWithEnumPrimaryKey entity = new EntityWithEnumPrimaryKey();
+					entity.setId( enumVal );
+					em.persist( entity );
+					entitiesWithEnumPk.add( entity );
+
+					em.flush();
+
+					// Native query is used here because HHH-15842 could cause persisting that test entity to fail
+					em.createNativeQuery( "INSERT INTO test_entity_enum_fk (id, enum_fk) VALUES (?1, ?2);" )
+							.setParameter( 1, UUID.randomUUID() )
+							.setParameter( 2, enumVal.name() )
+							.executeUpdate();
+				} ) );
+	}
+
+	@AfterEach
+	public void tearDown(EntityManagerFactoryScope scope) {
+		entitiesWithEnumPk.clear();
+		scope.inTransaction( em -> {
+			em.createQuery( "delete from EntityWithEnumForeignKey" ).executeUpdate();
+			em.createQuery( "delete from EntityWithEnumPrimaryKey" ).executeUpdate();
+		} );
+	}
+
+	@Test
+	public void testPersist(EntityManagerFactoryScope scope) {
+		scope.inTransaction( em -> {
+			for ( EntityWithEnumPrimaryKey entityWithEnumPk : entitiesWithEnumPk ) {
+				EntityWithEnumForeignKey entityWithFk = new EntityWithEnumForeignKey();
+				entityWithFk.setId( UUID.randomUUID() );
+				entityWithFk.setEntityWithEnumPk( entityWithEnumPk );
+				em.persist( entityWithFk );
+			}
+		} );
+	}
+
+	@Test
+	public void testQueryByRelatedEntity(EntityManagerFactoryScope scope) {
+		scope.inEntityManager( em -> {
+			for ( EntityWithEnumPrimaryKey entityWithEnumPk : entitiesWithEnumPk ) {
+				var q = em.createQuery(
+						"select e from EntityWithEnumForeignKey e where e.entityWithEnumPk = ?1",
+						EntityWithEnumForeignKey.class
+				).setParameter( 1, entityWithEnumPk );
+
+				assertThat( q.getResultList() ).hasSize( 1 );
+			}
+		} );
+	}
+
+	@Test
+	public void testQueryByRelatedEntityId(EntityManagerFactoryScope scope) {
+		scope.inEntityManager( em -> {
+			for ( EnumPK enumVal : EnumPK.values() ) {
+				var q = em.createQuery(
+						"select e from EntityWithEnumForeignKey e where e.entityWithEnumPk.id = ?1",
+						EntityWithEnumForeignKey.class
+				).setParameter( 1, enumVal );
+
+				assertThat( q.getResultList() ).hasSize( 1 );
+			}
+		} );
+	}
+
+	public enum EnumPK {
+		A, B
+	}
+
+	@Entity(name = "EntityWithEnumForeignKey")
+	@Table(name = "test_entity_enum_fk")
+	public static class EntityWithEnumForeignKey {
+
+		@Id
+		@Column(name = "id")
+		private UUID id;
```

Comment:
Please replace uses of `UUID` with `String` or `Long`. Sybase has some issues with trailing zeros and random UUIDs in other tests already cause intermittent failures. We don't need another test that does the same thing ;)